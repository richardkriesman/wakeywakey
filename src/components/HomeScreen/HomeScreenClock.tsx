import React, { ReactNode } from "react";
import { View, StyleSheet, ViewStyle } from "react-native";

import { Text } from "react-native";

export interface HomeScreenClockProps {
    wrapperStyle?   : ViewStyle;
}

/**
 * The clock on the home screen.
 * 
 * @author Shawn Lutch
 */
export class HomeScreenClock extends React.Component<HomeScreenClockProps> {
    constructor(props : HomeScreenClockProps) {
        super(props);
    }

    _pad(num : number) : string {
        return (num < 10 ? '0' : '') + num;
    }

    render() : ReactNode {
        let now = new Date();

        return (
            <View style={this.props.wrapperStyle}>
                <Text style={styles.clock}>
                    {this._pad(now.getHours())}:{this._pad(now.getMinutes())}
                </Text>
            </View>
        );
    }
};

const styles = StyleSheet.create({
    clock: {
        color: 'black',
        fontSize: 70,
        marginTop: 20,
        marginBottom: 20,
        textAlign: 'center'
    }
});
