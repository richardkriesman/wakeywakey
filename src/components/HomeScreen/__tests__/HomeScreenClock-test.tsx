import * as React from "react";
import renderer from "react-test-renderer";
import { HomeScreenClock } from "../HomeScreenClock";

import MockDate from "mockdate";

describe("HomeScreenClock", () => {

    jest.useFakeTimers();
    const fakeDateMillis = 1551896555862;

    beforeEach(() => {
        MockDate.set(fakeDateMillis);
    });

    it("renders the correct (fake) initial time", () => {
        const tree = renderer.create(<HomeScreenClock/>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("renders the correct (fake) later time", () => {
        MockDate.set(fakeDateMillis + 60000);
        const tree = renderer.create(<HomeScreenClock/>).toJSON();
        expect(tree).toMatchSnapshot();
    });

});
