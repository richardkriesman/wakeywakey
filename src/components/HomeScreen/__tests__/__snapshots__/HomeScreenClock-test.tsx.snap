// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreenClock renders the correct (fake) initial time 1`] = `
<View>
  <View
    style={
      Object {
        "marginBottom": 20,
        "marginTop": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 60,
          "textAlign": "center",
        }
      }
    >
      12:22 pm
    </Text>
  </View>
</View>
`;

exports[`HomeScreenClock renders the correct (fake) initial timing 1`] = `
<View>
  <View
    style={
      Object {
        "marginBottom": 20,
        "marginTop": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 60,
          "textAlign": "center",
        }
      }
    >
      12:22 pm
    </Text>
  </View>
</View>
`;

exports[`HomeScreenClock renders the correct (fake) later time 1`] = `
<View>
  <View
    style={
      Object {
        "marginBottom": 20,
        "marginTop": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 60,
          "textAlign": "center",
        }
      }
    >
      12:23 pm
    </Text>
  </View>
</View>
`;

exports[`HomeScreenClock renders the correct (fake) later timing 1`] = `
<View>
  <View
    style={
      Object {
        "marginBottom": 20,
        "marginTop": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 60,
          "textAlign": "center",
        }
      }
    >
      12:23 pm
    </Text>
  </View>
</View>
`;
